src/pages/home/report/ReportActionItemParentAction.tsx

```diff
@@ -1,91 +1,105 @@
                 <AnimatedEmptyStateBackground />
                 <View style={[styles.p5, StyleUtils.getReportWelcomeTopMarginStyle(isSmallScreenWidth)]} />
-                {parentReportAction && (
-                    <ReportActionItem
-                        // @ts-expect-error TODO: Remove the comment after ReportActionItem is migrated to TypeScript.
-                        report={report}
-                        action={parentReportAction}
-                        displayAsGroup={false}
-                        isMostRecentIOUReportAction={false}
-                        shouldDisplayNewMarker={shouldDisplayNewMarker}
-                        index={index}
-                    />
-                )}
+                {allAncestors.map((ancestor) => (
+                    <OfflineWithFeedback
+                        key={ancestor.reportAction.reportActionID}
+                        shouldDisableOpacity={Boolean(ancestor.reportAction?.pendingAction)}
+                        pendingAction={ancestor.report?.pendingFields?.addWorkspaceRoom ?? ancestor.report?.pendingFields?.createChat}
+                        errors={ancestor.report?.errorFields?.addWorkspaceRoom ?? ancestor.report?.errorFields?.createChat}
+                        errorRowStyles={[styles.ml10, styles.mr2]}
+                        onClose={() => Report.navigateToConciergeChatAndDeleteReport(ancestor.report.reportID)}
+                    >
+                        <ReportActionItem
+                            // @ts-expect-error TODO: Remove this once ReportActionItem (https://github.com/Expensify/App/issues/31982) is migrated to TypeScript.
+                            onPress={() => Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(ancestor.report.reportID))}
+                            report={ancestor.report}
+                            action={ancestor.reportAction}
+                            displayAsGroup={false}
+                            isMostRecentIOUReportAction={false}
+                            shouldDisplayNewMarker={ancestor.shouldDisplayNewMarker}
+                            index={index}
+                        />
+                        {!ancestor.shouldHideThreadDividerLine && <View style={[styles.threadDividerLine]} />}
+                    </OfflineWithFeedback>
+                ))}
             </View>
-            {!shouldHideThreadDividerLine && <View style={[styles.threadDividerLine]} />}
-        </OfflineWithFeedback>
+        </>
     );
 }
 
 ReportActionItemParentAction.displayName = 'ReportActionItemParentAction';
 
 export default withOnyx<ReportActionItemParentActionProps, ReportActionItemParentActionOnyxProps>({
-    report: {
-        key: ({reportID}) => `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+    // We should subscribe all reports and report actions here to dynamic update when any parent report action is changed
+    allReportActions: {
+        key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
```

Comment:
Hmm - this seems like it would introduce some performance trouble to me. Every action will subscribe to all other report action changes... I can't imagine that being good for performance. Did we test this on a chat that had many many report actions? If not - please add this to the QA test steps.